donut/modules/editor/static/script.js

```diff
@@ -0,0 +1,197 @@
+var old_title;
+// In milli
+let KEEP_PAGE_ALIVE_TIME = 60*1000;
+$(function() {	
+  old_title = $('#text_title').val();
+  $('#runBtn').click(preview);
+  $('#change_title').click(change_title);
+  for (var i = 1; i<7; i++) {
+    $('#heading'+String(i)).click({size:i}, insert_heading);
+  }
+  $('#italicize').click(insert_italic);
+  $('#bold').click(insert_bold);
+  $('#link').click(insert_link);
+  $('#image').click(insert_image);
+  $('#ulist').click(insert_ulist);
+  $('#olist').click(insert_olist);
+  $('#save').click(save);
+});
+
+// Creates html
+function preview() {
+  var text = $('#source').val(),
+      title = $('#text_title').val(),
+      target = $('#preview'),
+      target_title = $('#preview_title'),
+      converter = new showdown.Converter({strikethrough: true}),
+      html = converter.makeHtml(text);
+  target.html(html);
+  target_title.html(title);
```

Comment:
Use `.text()` if you don't need to insert arbitrary HTML strings. This makes it less likely that there will be a cross-site scripting attack possible

Reply:
We are inserting html

Reply:
Into the title, though?

Reply:
woops you right